components/director/internal/packagetobundles/handler.go

```diff
@@ -74,25 +77,11 @@ func (h *Handler) Handle() func(next http.Handler) http.Handler {
 					log.C(ctx).WithError(err).Error("Error determining request consumer")
 					appErr := apperrors.InternalErrorFrom(err, "while determining request consumer")
 					writeAppError(ctx, w, appErr, http.StatusInternalServerError)
-					return
 				}
 
-				if consumerInfo.ConsumerType == consumer.Runtime {
-					tenantID, err := tenant.LoadFromContext(ctx)
-					if err != nil {
-						log.C(ctx).WithError(err).Error("Error determining request tenant")
-						appErr := apperrors.InternalErrorFrom(err, "while determining request tenant")
-						writeAppError(ctx, w, appErr, http.StatusInternalServerError)
-						return
-					}
-
-					log.C(ctx).Infof("Proceeding with labeling runtime with ID %q with label %q", consumerInfo.ConsumerID, useBundlesParam)
-					h.labelUpsertService.UpsertLabel(ctx, tenantID, &model.LabelInput{
-						Key:        useBundlesParam,
-						Value:      "true",
-						ObjectID:   consumerInfo.ConsumerID,
-						ObjectType: model.LabelableObject(consumerInfo.ConsumerType),
-					})
+				if err := h.reconcileConsumerLabel(ctx, consumerInfo); err != nil {
+					writeAppError(ctx, w, err, http.StatusInternalServerError)
+					return
 				}
 
 				log.C(ctx).Infof("Returning request without rewriting the request body for consumer with ID %q and type %q", consumerInfo.ConsumerID, consumerInfo.ConsumerType)
```

Comment:
We can get the consumerInfo before `next.ServeHTTP(w, r)` and enrich the log on the first line to indicate which consumer has already adopted Bundles and then remove this log, since it does not give much of a value.

Reply:
Addressed